runtime/vm/resolvesupport.cpp

```diff
@@ -259,7 +259,14 @@ resolveClassRef(J9VMThread *vmStruct, J9ConstantPool *ramCP, UDATA cpIndex, UDAT
 	ramClassRefWrapper = (J9RAMClassRef *)&ramCP[cpIndex];
 	resolvedClass = ramClassRefWrapper->value;
 	/* If resolving for "new", check if the class is instantiable */
-	if ((NULL != resolvedClass) && (J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_INSTANTIABLE) || J9ROMCLASS_ALLOCATES_VIA_NEW(resolvedClass->romClass))) {
+	if ((NULL != resolvedClass) && (J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_INSTANTIABLE) ||
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+                (J9ROMCLASS_ALLOCATES_VIA_NEW(resolvedClass->romClass) || 
+                 (J9_IS_J9CLASS_VALUETYPE(resolvedClass) && J9_ARE_ANY_BITS_SET(resolvedFlags, J9_RESOLVE_FLAG_VALUETYPE)))
```

Comment:
use J9_ARE_ALL_BITS_SET

Reply:
also the || should be on the next line. Look at the coding guidelines to see how to do this properly